core/include/detray/materials/detail/relativistic_quantities.hpp

```diff
@@ -131,6 +104,29 @@ struct relativistic_quantities {
         return compute_epsilon_per_length(molarElectronDensity) * thickness;
     }
 
+    // Calculate the bethe_log_term of bethe equation,
+    // where bethe_log_term = 1/2 ln ( 2m_e c^2 beta^2 gamma^2 W_max/ I^2)
+    DETRAY_HOST_DEVICE scalar_type
+    compute_bethe_log_term(const scalar_type I) const {
+        // u = 2 * m_e c^2* beta^2 * gamma^2
+        const scalar_t u{compute_mass_term(constant<scalar_t>::m_e)};
+        const scalar_type A = 0.5f * math::log(u * m_Wmax / (I * I));
+        return A;
+    }
+
+    // Calculate d(bethe_log_term)/dqop
+    // where dA/dqop = - 1 / (2 * qop) * [4 - W_max/ (gamma M c^2) ]
+    DETRAY_HOST_DEVICE scalar_type derive_bethe_log_term() const {
+        const scalar_type dAdqop =
+            -1 / (2 * m_qOverP) * (4 - m_Wmax / (m_gamma * m_mass));
+        return dAdqop;
+    }
+
+    // Retrun d(beta^2)/dqop = - 2beta^2 / (qop * gamma^2)
+    DETRAY_HOST_DEVICE scalar_type derive_beta2() const {
+        return -2.f * m_beta2 / (m_qOverP * m_gamma2);
```

Comment:
Could you add a few assert against floating point exceptions (div 0)?

Reply:
Done